### PR TITLE
BHBC-1979: Update Templates to Use New Identifiers

### PR DESCRIPTION
**Pull request by KjartanE**
# Overview

## Links to Jira tickets

- https://quartech.atlassian.net/browse/BHBC-1979

## Description of relevant changes

- {List all relevant changes, in particular anything that will help the reviewers test/verify this PR}

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [x] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [x] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: database/src/migrations/20221105123456_update_templates_ids.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    Elk Aerial SRB Recruit Composition Survey
+    Elk Aerial Non SRB Recruit Composition Survey
+    Deer Aerial Non SRB Recruit Composition Survey
+    Deer Ground Transect Recruit Composition Survey
+*/
+export async function down(knex: Knex): Promise<void> {
+  await knex.raw(``);
+}
+
+const permutateTemplates = async (
```
Comment (AlfredRosenthal) [inline database/src/migrations/20221105123456_update_templates_ids.ts]: might be nice to have this in a utils page if/ when new templates are added

**Comment by AlfredRosenthal** [comment]
Just want to make sure I understand the new versioning for validations. Each time we need to add/ make a change we'd copy the contents into a new folder named yyyymmdd and make the changes there?

**Comment by AlfredRosenthal** [comment]
I've tried uploading a few of the updated templates from confluence but they all fail at getting validation, is this to be expected?

**Comment by curtisupshall** [comment]
> I've tried uploading a few of the updated templates from confluence but they all fail at getting validation, is this to be expected?

I'm guessing that the templates on Confluence are missing the newly coined sims_name and sims_version properties, and will need to be updated. Try using a template from S3?

**Comment by AlfredRosenthal** [comment]
> > I've tried uploading a few of the updated templates from confluence but they all fail at getting validation, is this to be expected?
> 
> I'm guessing that the templates on Confluence are missing the newly coined sims_name and sims_version properties, and will need to be updated. Try using a template from S3?

The files I tried from Confluence have the new `sims_name` and `sims_version`. The files from S3 look like they have the old template IDs

**Review by AlfredRosenthal — DISMISSED** [review]
🥼

File: api/src/repositories/validation-repository.test.ts [hunk 2]
```diff
@@ -16,74 +15,94 @@ describe('ValidationRepository', () => {
...
+        templateVersion,
+        fieldMethodId,
+        surveySpecies
+      );
+      expect(response.template_methodology_species_id).to.be.eql(1);
+      expect(response.validation).to.be.eql('{}');
+      expect(response.transform).to.be.eql('{}');
     });
 
     it('should throw `Failed to build SQL` error', async () => {
```
Comment (curtisupshall) [inline api/src/repositories/validation-repository.test.ts]: Should we rename this test?

File: database/src/migrations/template_methodology_species_validations/20221021/deer_aerial_non_srb_recruit_comp_survey.ts [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+          description: '',
+          validations: basicCodeValidator(DeerItems.latLongUnits)
+        },
+        {
+          name: 'Adult Males',
+          description: '',
+          validations: basicNumericValidator()
+        },
+        {
+          name: 'Adilt Females',
```
Comment (curtisupshall) [inline database/src/migrations/template_methodology_species_validations/20221021/deer_aerial_non_srb_recruit_comp_survey.ts]: Typo.

File: database/src/migrations/template_methodology_species_validations/20221021/deer_ground_transect_recruit_comp_survey.ts [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+          description: '',
+          validations: basicCodeValidator(DeerItems.latLongUnits)
+        },
+        {
+          name: 'Adult Males',
+          description: '',
+          validations: basicNumericValidator()
+        },
+        {
+          name: 'Adilt Females',
```
Comment (curtisupshall) [inline database/src/migrations/template_methodology_species_validations/20221021/deer_ground_transect_recruit_comp_survey.ts]: Repeat typo.

File: database/src/migrations/template_methodology_species_validations/20221021/elk_non_srb_survey.ts [hunk 5]
```diff
@@ -0,0 +1,601 @@
...
+          description: '',
+          validations: basicCodeValidator(ElkItems.latLongUnits)
+        },
+        {
+          name: 'Adult Males',
+          description: '',
+          validations: basicNumericValidator()
+        },
+        {
+          name: 'Adilt Females',
```
Comment (curtisupshall) [inline database/src/migrations/template_methodology_species_validations/20221021/elk_non_srb_survey.ts]: Repeat typo.

File: database/src/migrations/template_methodology_species_validations/20221021/picklist.ts [hunk 6]
```diff
@@ -0,0 +1,545 @@
+/* eslint-disable prettier/prettier */
+const yN = [
+  { name: 'Y', description: '' },
+  { name: 'N', description: '' }
+];
+
+const yND = [
+  { name: 'Y', description: 'Yes' },
+  { name: 'N', description: 'No' },
+  { name: "Didn''t Listen", description: "Didn''t Listen" }
```
Comment (curtisupshall) [inline database/src/migrations/template_methodology_species_validations/20221021/picklist.ts]: Typo: double apostrophe

**Review by curtisupshall — COMMENTED** [review]
Looks good! Just noticed some typos 🍪

**Review by curtisupshall — APPROVED** [review]

**Pull request merged**